app/controllers/api/irs_attempts_api_controller.rb

```diff
@@ -42,6 +47,15 @@ def security_event_tokens
       @security_event_tokens ||= redis_client.read_events(timestamp: timestamp)
     end
 
+    def encrypted_security_event_log_result
+      json = security_event_tokens.to_json
+      gzip = Zlib.gzip(json)
+      decoded_key_der = Base64.strict_decode64(IdentityConfig.store.irs_attempt_api_public_key)
+      pub_key = OpenSSL::PKey::RSA.new(decoded_key_der)
+
+      IrsAttemptsApi::Encryptor.new(data: gzip, timestamp: timestamp, public_key: pub_key).encrypt
```

Comment:
one more idea: would it make sense to include the GZipping inside the encryptor/batch encryptor class? So the batch encryptor takes an array of events and packages them all up?

Reply:
I went back and forth on that since it's not _strictly_ part of encryption (similar to the JSON encoding), but I'm not strongly tied to that.

Reply:
ok! more in the spirit of getting logic out of controllers then? i also feel medium strong at best

Reply:
Yep, thanks for the nudge, moved in 974e9fac6